lstchain/io/dataselection.py

```diff
@@ -0,0 +1,248 @@
+        trait=Float(),
+        default_value=[0, np.inf],
+    ).tag(config=True)
+
+    width = List(
+        help="Range of width for event filter",
+        trait=Float(),
+        default_value=[0, np.inf],
+    ).tag(config=True)
+
+    r = List(
+        help="Range of r for event filter",
+        trait=Float(),
+        default_value=[0, 1],
+    ).tag(config=True)
+
+    wl = List(
+        help="Range of wl for event filter",
+        trait=Float(),
+        default_value=[0.01, 1],
+    ).tag(config=True)
+
+    leakage_intensity_width_2 = List(
+        help="Range for leakage_intensity_width_2 of event filter",
+        trait=Float(),
+        default_value=[0, 1],
+    ).tag(config=True)
+
+    fixed_gh_cut = Float(
+        help="Fixed selection cut for gh_score (gammaness)",
+        default_value=0.5,
+    ).tag(config=True)
+
+    fixed_theta_cut = Float(
+        help="Fixed selection cut for theta",
+        default_value=0.5,
+    ).tag(config=True)
+
+    fixed_source_fov_offset_cut = Float(
+        help="Fixed selection cut for source FoV offset",
+        default_value=3,
+    ).tag(config=True)
+
+    lst_tel_ids = List(
+        help="List of selected LST telescope ids",
+        trait=Int(),
+        default_value=[1],
+    ).tag(config=True)
+
+    def __init__(self, parent=None, **kwargs):
+        super().__init__(parent=parent, **kwargs)
+
+    def __call__(self, data):
+        """
+        Check if the table passed is QTable or not
+        """
+        if type(data).__name__ != 'QTable':
+            self.log.debug("Data table is not of QTable class")
+            pass
+
+    def filter_cut(self, data):
+        return filter_events(data, self.event_filters())
+
+    def gh_cut(self, data):
+        return data[data["gh_score"] > self.fixed_gh_cut]
+
+    def theta_cut(self, data):
+        return data[data["theta"] < u.Quantity(self.fixed_theta_cut) * u.deg]
+
+    def true_src_fov_offset_cut(self, data):
+        return data[
+                data["true_source_fov_offset"] < u.Quantity(
+                    self.fixed_source_fov_offset_cut) * u.deg
+            ]
+
+    def reco_src_fov_offset_cut(self, data):
+        return data[
+                data["reco_source_fov_offset"] < u.Quantity(
+                    self.fixed_source_fov_offset_cut) * u.deg
+            ]
+
+    def tel_ids_filter(self, data):
+        for i in self.lst_tel_ids:
+            data["sel_tel"] = data["tel_id"] == i
+        return data[data["sel_tel"]]
+
+    def event_filters(self):
+        """
+        Creates a dict for filter_events() function
+        """
+        self.evt_filter = {
+            "intensity": self.intensity,
+            "width": self.width,
+            "length": self.length,
+            "r": self.r,
+            "wl": self.wl,
+            "leakage_intensity_width_2": self.leakage_intensity_width_2,
+        }
+        return self.evt_filter
+
+
+class DataBinning(Component):
```

Comment:
You again duplicate things are and are inconsistent with what you give.

I would propose to either give the bin edges (most flexible but terrible to type manually), or the min, max, n_bins for all.

You also have multiple versions of `fov_offset_bins`, but only one should be needed and the content is then deciding of wether it is multiple or not.

Reply:
Providing the bin edges will be a lot to provide. But so will be individual min, max and n_bins for all these parameters! With lists like these, I understand that it is not intuitive, but it seems easier to enter the values, even in a config file.

I do not understand how this is duplicating things. Can you please clarify?

For `fov_offset_bins` I  used a different for background IRF, and for point-like or full enclosure IRFs, yes, I can make it only a single traitlet and then let the code check, but for now, I am just letting there be only 1 check on the flag `point-like` for `lstchain_create_irf_files` and then the Tool selects the appropriate `fov_offset_bins` either provided or the default value

Reply:
There is a traitlet called `single_offset_fov_bins` and one called `multiple_offset_fov_bins`. I don't see why both of these are needed.